src/map/atcommand.c

```diff
@@ -8544,6 +8544,9 @@ ACMD(set)
 			data->type = C_STR;
 			data->u.mutstr = aStrdup(str);
 		}
+	} else if (script->str_data[script->add_str(reg)].type == C_PARAM) {
```

Comment:
Better not add all input strings into db,
try replace script->add_str to script->search_str